### ISSUE TITLE
Inconsistent results for certain strings?

### ISSUE THREAD

**Issue by philbooth** [philbooth/surch#1]
Depending on what other data has been added to an index, I'm seeing certain strings fail to match incorrectly. Working on a minimal test case now.

**Comment by philbooth** [comment]
The problem:

```js
const index = surch.create('foo');
index.add({ _id: '01', foo: 'The Queen\'s Head' });
index.add({ _id: '02', foo: 'The Craft Beer Co.' });

// Returns an empty array!
const result = index.search('The Craft Beer Co.');
```

No problem:

```js
const index = surch.create('foo');
index.add({ _id: '02', foo: 'The Craft Beer Co.' });

// Returns the correct result!
const result = index.search('The Craft Beer Co.');
```

**Comment by philbooth** [comment]
Further reduced:

```js
const index = surch.create('foo');
index.add({ _id: '01', foo: 'The Quee' });
index.add({ _id: '02', foo: 'The Beer' });

// Returns an empty array!
const result = index.search('The Beer');
```

**Comment by philbooth** [comment]
Fixed in f3ee722, shipped in `0.2.1`.
